### PR TITLE
Move runSafeCommand to console.RunSafeCommand

### PR DESCRIPTION
**Pull request by qinqon**
Signed-off-by: Quique Llorente <ellorent@redhat.com>

**What this PR does / why we need it**:
The multus tests has a function to pass the command to execute and
execute them at the console ensuring that commands are returning ok,
this change move that functions to "console" package so other tests can
use it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by maiqueb — CHANGES_REQUESTED** [review]
I'm OK with this, but for this to be a follow-up of #4923 , it should be rebased of top of it, right ?

Furthermore, you're doing 2 things here: export the function to the console pkg, *and* changing the API of one public method. The latter is causing you to change tests throughout the code base, which is unrelated to the change. I would prefer if you split these into 2 different commits.

Lastly, I'm a bit unsure of the exported command name. Safe command? Imo, it would be better if it hinted that the command must be successful (have ret code 0). Probably that's what 'safe' is attempting to convey, but it doesn't feel right. `ExecuteSuccessfulCmd` ? I don't have a better naming suggestion unfortunately.

Does it make sense to you ?

**Comment by qinqon** [comment]
> I'm OK with this, but for this to be a follow-up of #4923 , it should be rebased of top of it, right ?

Sure, will rebase this when the #4923 is merged.

> Furthermore, you're doing 2 things here: export the function to the console pkg, _and_ changing the API of one public method. The latter is causing you to change tests throughout the code base, which is unrelated to the change. I would prefer if you split these into 2 different commits.

Will do.
 
> Lastly, I'm a bit unsure of the exported command name. Safe command? Imo, it would be better if it hinted that the command must be successful (have ret code 0). Probably that's what 'safe' is attempting to convey, but it doesn't feel right. `ExecuteSuccessfulCmd` ? I don't have a better naming suggestion unfortunately.
> 

It's not about ret code, what it does is to check that command are _really_ sended to the console waiting for them to appear with an expect, so `Safe` is good name.

**Comment by maiqueb** [comment]
> > I'm OK with this, but for this to be a follow-up of #4923 , it should be rebased of top of it, right ?
> 
> Sure, will rebase this when the #4923 is merged.
> 
> > Furthermore, you're doing 2 things here: export the function to the console pkg, _and_ changing the API of one public method. The latter is causing you to change tests throughout the code base, which is unrelated to the change. I would prefer if you split these into 2 different commits.
> 
> Will do.
> 
> > Lastly, I'm a bit unsure of the exported command name. Safe command? Imo, it would be better if it hinted that the command must be successful (have ret code 0). Probably that's what 'safe' is attempting to convey, but it doesn't feel right. `ExecuteSuccessfulCmd` ? I don't have a better naming suggestion unfortunately.
> 
> It's not about ret code, what it does is to check that command are _really_ sended to the console waiting for them to appear with an expect, so `Safe` is good name.

I mean that the command asserts that the function's return code is 0. I'd like the function's name to somehow reflect that.

**Comment by qinqon** [comment]
> > > I'm OK with this, but for this to be a follow-up of #4923 , it should be rebased of top of it, right ?
> > 
> > 
> > Sure, will rebase this when the #4923 is merged.
> > > Furthermore, you're doing 2 things here: export the function to the console pkg, _and_ changing the API of one public method. The latter is causing you to change tests throughout the code base, which is unrelated to the change. I would prefer if you split these into 2 different commits.
> > 
> > 
> > Will do.
> > > Lastly, I'm a bit unsure of the exported command name. Safe command? Imo, it would be better if it hinted that the command must be successful (have ret code 0). Probably that's what 'safe' is attempting to convey, but it doesn't feel right. `ExecuteSuccessfulCmd` ? I don't have a better naming suggestion unfortunately.
> > 
> > 
> > It's not about ret code, what it does is to check that command are _really_ sended to the console waiting for them to appear with an expect, so `Safe` is good name.
> 
> I mean that the command asserts that the function's return code is 0. I'd like the function's name to somehow reflect that.

Well, `RunCommand` fails if the command is not success, I think that what we can do is to keep the name and to document when do we return and error.

**Comment by maiqueb** [comment]
> > > > I'm OK with this, but for this to be a follow-up of #4923 , it should be rebased of top of it, right ?
> > > 
> > > 
> > > Sure, will rebase this when the #4923 is merged.
> > > > Furthermore, you're doing 2 things here: export the function to the console pkg, _and_ changing the API of one public method. The latter is causing you to change tests throughout the code base, which is unrelated to the change. I would prefer if you split these into 2 different commits.
> > > 
> > > 
> > > Will do.
> > > > Lastly, I'm a bit unsure of the exported command name. Safe command? Imo, it would be better if it hinted that the command must be successful (have ret code 0). Probably that's what 'safe' is attempting to convey, but it doesn't feel right. `ExecuteSuccessfulCmd` ? I don't have a better naming suggestion unfortunately.
> > > 
> > > 
> > > It's not about ret code, what it does is to check that command are _really_ sended to the console waiting for them to appear with an expect, so `Safe` is good name.
> > 
> > 
> > I mean that the command asserts that the function's return code is 0. I'd like the function's name to somehow reflect that.
> 
> Well, `RunCommand` fails if the command is not success, I think that what we can do is to keep the name and to document when do we return and error.

I don't think those things are mutually exclusive; you can document the function, and you can (and should) name it properly so anyone lazy enough (like me) to not read its doctext knows what it does.

**Comment by qinqon** [comment]
[force-push](https://github.com/kubevirt/kubevirt/compare/c5de9f19b7ca5cc87d8e321b02d1c321c15cd6f5..def74b9bbe8c3a7b99922fa5cc86410c50e51fdc): Address Miguel's comments

File: tests/console/console.go [hunk 1]
```diff
@@ -128,6 +128,20 @@ func RunCommand(vmi *v1.VirtualMachineInstance, command string, timeout time.Dur
 	return nil
 }
 
+// RunSafeCommandAndSucceed runs the command line from `command connecting to an already logged in console at vmi
+// and wait `timeout` for command to return, it uses SafeExpectBatch so the command is checked with expect at
+// the console. It will return an error if command does not success or command does not appear at console
```
Comment (maiqueb) [inline tests/console/console.go:133]: typo: "It will return an error if command does not **succeed**"

**Comment by maiqueb** [comment]
/lgtm
/hold

Let's have https://github.com/kubevirt/kubevirt/pull/4923 merged first.

**Comment by maiqueb** [comment]
/hold cancel

#4923 was merged, you should rebase this.

Thanks!

File: tests/console/console.go [hunk 2]
```diff
@@ -128,6 +128,20 @@ func RunCommand(vmi *v1.VirtualMachineInstance, command string, timeout time.Dur
 	return nil
 }
 
+// RunSafeCommandAndSucceed runs the command line from `command connecting to an already logged in console at vmi
```
Comment (dankenigsberg) [inline tests/console/console.go:131]: the name in comment is inconsistent with the real one (that's always a problem with comments!)

also, the name `RunSafeCommandSuccessfully` is not clear to me. I don't know why it is "safe" nor "successful". 
How do you feel about `RunConsoleCommand`? `RunAndCheckShellCommand`?

Comment (qinqon) [reply tests/console/console.go:131]: We have documented what the command means in the godoc and we express what "Safe" means and also we know what Successfully means implcitly, so I think this is ok.

I will change the commit comments though, thanks for review.

Comment (dankenigsberg) [reply tests/console/console.go:131]: I may well be mistaken, as I'm not familiar with the function and its usages. I'm not holding this PR. From elsewhere in this file it seems that  "Safe" means "ValidatedSend". Is that so? Does the standard library or another library you value have `Successfully` is a function name? `console.RunConsoleCommand` is indeed bad.

BTW should the backtick before "command" be closed?

Comment (maiqueb) [reply tests/console/console.go:131]: > I may well be mistaken, as I'm not familiar with the function and its usages. I'm not holding this PR. From elsewhere in this file it seems that "Safe" means "ValidatedSend". Is that so? Does the standard library or another library you value have `Successfully` is a function name? `console.RunConsoleCommand` is indeed bad.
> 
> BTW should the backtick before "command" be closed?

I associate "success" while executing a command with return code 0, which is something asserted within that function. That was the reason why I insisted in adding that to the function name. 

Can't venture into the "safe" part.

Comment (qinqon) [reply tests/console/console.go:131]: What about `console.RunCommandAndValidateSend` ?

Comment (maiqueb) [reply tests/console/console.go:131]: Reading this again, I'm forced to agree this method's name is confusing. 

I don't know what the safe regards to.

**Review by maiqueb — COMMENTED** [review]
I wonder about the first commit in this PR; you export a function that is locally available to a pkg, then use it inside of that package. What's the point of exposing it in the first place ? No one uses it. 

Did I miss something ?

**Pull request closed without merging**